gnomad_qc/v3/annotations/generate_freq_data.py

```diff
@@ -1,104 +1,270 @@
+                (freq_ht[mt.row_key].AF > 0.01)
+                & mt.GT.is_het()
+                & (mt.AD[1] / mt.DP > 0.9),
+                hl.call(1, 1),
+                mt.GT,
+            )
+        )
 
-    logger.info("Generating frequency data...")
-    mt = annotate_freq(
-        mt,
-        sex_expr=mt.meta.sex,
-        pop_expr=mt.meta.pop
-    )
+        logger.info("Generating frequency data...")
+        if subset:
+            mt = annotate_freq(
+                mt,
+                sex_expr=mt.meta.sex_imputation.sex_karyotype,
+                pop_expr=mt.meta.population_inference.pop
+                if subset not in COHORTS_WITH_POP_STORED_AS_SUBPOP
+                else mt.meta.project_meta.project_subpop,
+                # NOTE: TGP and HGDP labeled populations are highly specific and are stored in the project_subpop meta field
+            )
 
-    # Select freq, FAF and popmax
-    faf, faf_meta = faf_expr(mt.freq, mt.freq_meta, mt.locus, POPS_TO_REMOVE_FOR_POPMAX)
-    mt = mt.select_rows(
-        'InbreedingCoeff',
-        'freq',
-        faf=faf,
-        popmax=pop_max_expr(mt.freq, mt.freq_meta, POPS_TO_REMOVE_FOR_POPMAX)
-    )
-    mt = mt.annotate_globals(faf_meta=faf_meta)
+            # NOTE: no FAFs or popmax needed for subsets
+            mt = mt.select_rows("freq")
 
-    # Annotate quality metrics histograms, as these also require densifying
-    mt = mt.annotate_rows(
-        **qual_hist_expr(mt.GT, mt.GQ, mt.DP, mt.AD)
-    )
+            logger.info(f"Writing out frequency data for {subset} subset...")
+            if args.test:
+                mt.rows().write(
+                    f"gs://gnomad-tmp/gnomad_freq/chr20_test_freq.{subset}.ht",
+                    overwrite=True,
+                )
+            else:
+                mt.rows().write(get_freq(subset=subset).path, overwrite=args.overwrite)
 
-    logger.info("Writing out frequency data...")
-    if args.test:
-        mt.rows().write("gs://gnomad-tmp/gnomad_freq/chr20_1_1000000_freq.ht", overwrite=True)
-    else:
-        mt.rows().write(freq.path, overwrite=args.overwrite)
+        else:
+            logger.info("Computing age histograms for each variant...")
+            mt = mt.annotate_cols(
+                age=hl.if_else(
+                    hl.is_defined(mt.meta.project_meta.age),
+                    mt.meta.project_meta.age,
+                    mt.meta.project_meta.age_alt,
+                    # NOTE: most age data is stored as integers in 'age' annotation, but for a select number of samples, age is stored as a bin range and 'age_alt' corresponds to an integer in the middle of the bin
+                )
+            )
+            mt = mt.annotate_rows(
+                **age_hists_expr(
+                    mt.adj,
+                    mt.GT,
+                    mt.age,
+                )
+            )
+
+            # Compute callset-wide age histogram global
+            meta_ht = meta.ht().select('release')
```

Comment:
I think this block can be simplified because mt already has the meta info and is already filtered to release samples by: 
```        mt = get_gnomad_v3_mt(
            key_by_locus_and_alleles=True, release_only=True, samples_meta=True
        )
```
though I might just be missing something. 

So I think it can just be:

```
mt = mt.annotate_globals(
    age_distribution=mt.aggregate_cols(
        hl.agg.hist(
            mt.age,
            30,
            80,
            10,
        )
    )
)
```

Reply:
Good catch!